File: comportamiento/memento/ejemplo.py
class Memento:
    def __init__(self, state): self._state = state
    def get_state(self): return self._state

class Originator:
    state = ""
    def save(self): return Memento(self.state)
    def restore(self, m): self.state = m.get_state()

File: comportamiento/memento/ejemplo.java
class Memento {
    private String s;
    public Memento(String s) { this.s = s; }
    public String getState() { return s; }
}
class Originator {
    String state;
    public Memento save() { return new Memento(state); }
    public void restore(Memento m) { state = m.getState(); }
}